### DISCUSSION TITLE
WebMidiJS and Electron

### DISCUSSION THREAD

**Discussion by localhero44** [djipco/webmidi#302] [Q&A] [answered]
Hi,

Thanks for this nice library that I use for several months (version 3.0.21 now) :)
When used on a web page it works perfectly connected to a Teensy.
Recently I have tried to transform my page into a standalone app with Electron.
When I launch the Electron app, I see WebMIDI enabled, but the inputs/outputs arrays are empty.
What am I doing wrong ?
I use the same code than the one provided as example.

David

**Comment by localhero44** [comment, marked as answer]
Stupid me, the web page was also running, so the MIDI device was used by the web page :(
Now I see the device with WebMIDI in the Electron app.
It's not intuitive to understand, WebMIDI enabled but no input/output found.

**Reply by djipco** [reply]
Hmm... interesting. Is this on Windows? I believe this is not a problem on macOS. In any case, I'm not sure what I could do to help the user in such a scenario.

**Reply by rubendax** [reply]
Yes, unfortunately [this is an issue built into Windows OS](https://tristancalderbank.com/2020/08/19/how-to-use-the-same-midi-device-on-windows-across-multiple-programs-at-the-same-time/). Windows only allows one application to receive MIDI at a time. Mac OS does not have this restriction.

**Reply by localhero44** [reply]
Hi, sorry long time answer, yes it's on Windows. If I don't close the web page lets's say on Chrome, then in another browser the device is not available for MIDI.

**Reply by djipco** [reply]
I wish I could help you but that seems to be the situation on Windows. 😕

**Comment by localhero44** [comment]
Bonsoir, oui bien sûr, ce n'était pas une demande, juste le constat que c'était bien un problème lié à Windows :)
